q2_fmt/tests/test_engraftment.py

```diff
@@ -488,10 +513,114 @@ def test_alpha_dists_with_extra_samples_in_diversity_not_in_metadata(self):
+                                    " the `distance_to` parameter and a"
+                                    " `baseline_timepoint` was not provided."):
+            group_timepoints(diversity_measure=self.alpha,
+                             metadata=self.md_alpha, distance_to='baseline',
+                             time_column='days_post_transplant')
+
+    def test_d2_baseline_alpha(self):
+        md_baseline = Metadata.load(self.get_data_path(
+                       'sample_metadata_a_div_baseline.txt'))
+
+        exp_time_df = pd.DataFrame({
+            'id': ['sampleD', 'sampleC', 'sampleE', 'sampleF'],
+            'measure': [6, 15, 44, 17,],
+            'group': [11.0, 9.0, 11.0, 9.0,],
+            'subject': ['subject1', 'subject2', 'subject2', 'subject2']
+        })
+
+        exp_ref_df = pd.DataFrame({
+            'id': ['sampleA', 'sampleB',
+                   'sampleC', 'sampleD', 'sampleE', 'sampleF'],
+            'measure': [24, 37, 15, 6, 44, 17],
+            'group': ['reference', 'reference',
+                      'control1', 'control1', 'control2', 'control2']
+        })
+
+        time_df, ref_df = group_timepoints(diversity_measure=self.alpha,
+                                           metadata=md_baseline,
+                                           distance_to='baseline',
+                                           time_column='days_post_transplant',
+                                           subject_column='subject',
+                                           baseline_timepoint=7.0,
+                                           control_column='control')
+
+        pd.testing.assert_frame_equal(time_df, exp_time_df)
+        pd.testing.assert_frame_equal(ref_df, exp_ref_df)
+
+    def test_d2_baseline_beta_dists(self):
+        exp_time_df = pd.DataFrame({
+            'id': ['sampleC', 'sampleD', 'sampleE'],
+            'measure': [0.52, 0.51, 0.36],
+            'group': [9.0, 11.0, 11.0],
+            'subject': ['subject1', 'subject1', 'subject2']
+        })
+
+        exp_ref_df = pd.DataFrame({
+            'id': ['sampleA..sampleB'],
+            'measure': [0.48],
+            'group': ['reference'],
+            'A': ['sampleA'],
+            'B': ['sampleB']
+        })
+
+        time_df, ref_df = group_timepoints(diversity_measure=self.dm,
+                                           metadata=self.md_beta,
+                                           distance_to='baseline',
+                                           subject_column='subject',
+                                           time_column='days_post_transplant',
+                                           baseline_timepoint=7)
+
+        pd.testing.assert_frame_equal(time_df, exp_time_df)
+        pd.testing.assert_frame_equal(ref_df, exp_ref_df)
+
+    def test_no_baseline_duplicates(self):
+        with self.assertRaisesRegex(ValueError, "More then one baseline sample"
```

Comment:
```suggestion
        with self.assertRaisesRegex(ValueError, "More than one baseline sample"
```
I double checked where you raise this error and you do have have the spelling correct there - so this should resolve a test failure due to mismatched regex (although we don't get to see any test failures on GH b/c of the old ci that's running).